other/fun/create_savedata.c

```diff
@@ -0,0 +1,115 @@
+/* SPDX-License-Identifier: GPL-3.0-or-later
+ *
+ * Creates Tox savedata for a given secret key, if provided, or a random key otherwise.
+ * The data is written to stderr, human-readable key info is written to stdout.
+ *
+ * Build: gcc -o create_savedata create_savedata.c -lsodium -ltoxcore -std=c99
+ *
+ * Usage: ./create_savedata [secret-key] 2>data
+ */
+#include <stdbool.h>
+#include <stdio.h>
+#include <stdlib.h>
+
+#include <sodium.h>
+
+#include "../../toxcore/tox.h"
+#include "create_common.h"
+
+static bool create_tox(const unsigned char *const secret_key, Tox **const tox)
+{
+    Tox_Err_Options_New options_error;
+    struct Tox_Options *const options = tox_options_new(&options_error);
+
+    if (options_error != TOX_ERR_OPTIONS_NEW_OK) {
+        tox_options_free(options);
+        return false;
+    }
+
+    tox_options_set_savedata_type(options, TOX_SAVEDATA_TYPE_SECRET_KEY);
+    tox_options_set_savedata_data(options, secret_key, crypto_box_SECRETKEYBYTES);
+    Tox_Err_New tox_error;
+    *tox = tox_new(options, &tox_error);
+
+    if (tox_error != TOX_ERR_NEW_OK) {
+        tox_options_free(options);
+        return false;
+    }
+
+    tox_options_free(options);
+    return true;
+}
+
+static bool print_savedata(Tox *const tox)
+{
+    const size_t savedata_size = tox_get_savedata_size(tox);
+    uint8_t *const savedata = (uint8_t *const)malloc(savedata_size);
+
+    if (savedata == nullptr) {
+        tox_kill(tox);
+        return false;
```

Comment:
`Memory leak: savedata` -- cppcheck

Reply:
Are you sure? `savedata` doesn't leak, it's `NULL` here.

The issue here is the `tox_kill(tox);` call I forgot to remove after all the refactoring I did, resulting in a double-kill. Fixing that now...